File: cf_bot/js_bot/handlers/menu.js
const { getUserLang } = require('../utils/session');
const { TEXTS } = require('../utils/texts');

/**
 * Setup menu buttons handlers
 * @param {import('telegraf').Telegraf} bot 
 */
function setupMenuHandlers(bot) {

    // Helper to get arrays of text representations for each feature across all languages
    const tenderBtns = [TEXTS.uz.btn_tender_uz, TEXTS.ru.btn_tender_uz, TEXTS.en.btn_tender_uz];
    const exhUzBtns = [TEXTS.uz.btn_exh_uz, TEXTS.ru.btn_exh_uz, TEXTS.en.btn_exh_uz];
    const exhGlbBtns = [TEXTS.uz.btn_exh_glb, TEXTS.ru.btn_exh_glb, TEXTS.en.btn_exh_glb];
    const productBtns = [TEXTS.uz.btn_products, TEXTS.ru.btn_products, TEXTS.en.btn_products];

    // Tender News
    bot.hears(tenderBtns, async (ctx) => {
        const lang = getUserLang(ctx.from.id);
        const text = TEXTS[lang].msg_tender_uz;
        await ctx.reply(text, { parse_mode: 'HTML' });
    });

    // Exhibitions UZ
    bot.hears(exhUzBtns, async (ctx) => {
        const lang = getUserLang(ctx.from.id);
        const text = TEXTS[lang].msg_exh_uz;
        await ctx.reply(text, { parse_mode: 'HTML' });
    });

    // Exhibitions Global
    bot.hears(exhGlbBtns, async (ctx) => {
        const lang = getUserLang(ctx.from.id);
        const text = TEXTS[lang].msg_exh_glb;
        await ctx.reply(text, { parse_mode: 'HTML' });
    });

    // Popular Products
    bot.hears(productBtns, async (ctx) => {
        const lang = getUserLang(ctx.from.id);
        const text = TEXTS[lang].msg_products;
        await ctx.reply(text, { parse_mode: 'HTML' });
    });

    // Fallback unknown command handling
    // Telegraf processes middlewares sequentially. This should be the last text match.
    // However, it's simpler to use bot.on('message') but it catches everything including photos.
    // We only react to text messages.
    bot.on('text', async (ctx, next) => {
        // If none of the 'hears' matched, this will execute
        const lang = getUserLang(ctx.from.id);
        const text = TEXTS[lang].unknown_command;
        await ctx.reply(text);
    });
}

module.exports = { setupMenuHandlers };

File: python_bot/handlers/menu.py
from aiogram import Router, F
from aiogram.types import Message

from utils.session import get_user_lang
from utils.texts import TEXTS

router = Router()

@router.message(F.text.in_([
    TEXTS['uz']['btn_tender_uz'], 
    TEXTS['ru']['btn_tender_uz'], 
    TEXTS['en']['btn_tender_uz']
]))
async def handle_tender_news(message: Message):
    """O'zbekiston tender yangiliklari tugmasi bosilganda"""
    lang = get_user_lang(message.from_user.id)
    text = TEXTS[lang]['msg_tender_uz']
    
    await message.answer(text, parse_mode="HTML")


@router.message(F.text.in_([
    TEXTS['uz']['btn_exh_uz'], 
    TEXTS['ru']['btn_exh_uz'], 
    TEXTS['en']['btn_exh_uz']
]))
async def handle_exhibitions_uz(message: Message):
    """O'zbekistondagi ko'rgazmalar tugmasi bosilganda"""
    lang = get_user_lang(message.from_user.id)
    text = TEXTS[lang]['msg_exh_uz']
    
    await message.answer(text, parse_mode="HTML")


@router.message(F.text.in_([
    TEXTS['uz']['btn_exh_glb'], 
    TEXTS['ru']['btn_exh_glb'], 
    TEXTS['en']['btn_exh_glb']
]))
async def handle_exhibitions_global(message: Message):
    """Dunyo bo'ylab ko'rgazmalar tugmasi bosilganda"""
    lang = get_user_lang(message.from_user.id)
    text = TEXTS[lang]['msg_exh_glb']
    
    await message.answer(text, parse_mode="HTML")


@router.message(F.text.in_([
    TEXTS['uz']['btn_products'], 
    TEXTS['ru']['btn_products'], 
    TEXTS['en']['btn_products']
]))
async def handle_popular_products(message: Message):
    """Eng ko'p qidirilgan mahsulotlar tugmasi bosilganda"""
    lang = get_user_lang(message.from_user.id)
    text = TEXTS[lang]['msg_products']
    
    await message.answer(text, parse_mode="HTML")


@router.message()
async def handle_unknown(message: Message):
    """Tushunarsiz buyruq yoki matn kelganda (fallback)"""
    lang = get_user_lang(message.from_user.id)
    text = TEXTS[lang]['unknown_command']
    
    await message.answer(text)
